tenma/tenmaDcLib.py

```diff
@@ -48,34 +50,47 @@ def instantiate_tenma_class_from_device_response(device, debug=False):
         The subclasses mainly deal with the limit checks for each
         unit.
     """
-    # Fist instantiate base to retrieve version
-    T = Tenma72Base(device, debug=debug)
-    ver = T.getVersion()
-    T.close()
-    
-    for cls in Tenma72Base.__subclasses__():
-        if cls.MATCH_STR in ver:
-            return cls(device, debug)
-
-    print("Could not detect Tenma Model, assuming 72_2545")
-    return Tenma72_2545(device, debug)
+    # First instantiate base to retrieve version
+    powerSupply = Tenma72Base(device, debug=debug)
+    ver = powerSupply.getVersion()
+    if not ver and debug:
```

Comment:
Be careful since this might not be expected, I think you want to retry even when not in debug mode (but only show the message if debug mode is enabled).

Reply:
Haha good catch